### PR TITLE
CSHARP-3078: Fix test CreateIndex_with_wildcardProjection_should_create_expected_index on latest

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
https://jira.mongodb.org/browse/CSHARP-3078
https://evergreen.mongodb.com/version/5ee7a0730305b963ecabedcc

### CODE REVIEW

File: tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs [hunk 1]
```diff
@@ -89,7 +89,14 @@ public void CreateIndex_with_wildcardProjection_should_create_expected_index()
             var indexes = collection.GetIndexes();
             var index = indexes.RawDocuments.Single(i => i["name"].AsString == "custom");
             index["key"]["$**"].AsInt32.Should().Be(1);
-            index["wildcardProjection"].ToBsonDocument().Should().Be(BsonDocument.Parse("{ b : 1, _id : 0 }"));
+            if (CoreTestConfiguration.ServerVersion >= new SemanticVersion(4, 5, 0, ""))
+            {
+                index["wildcardProjection"].ToBsonDocument().Should().Be(BsonDocument.Parse("{ b : true, _id : false }"));
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs]: just minor suggestion, you can move this code:

    index["wildcardProjection"].ToBsonDocument()

 into the separate variable before `if`

**Review by DmitryLukyanov — APPROVED** [review]
Minor suggestion and then LGTM

File: tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs (hunk 1 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs]: I've noticed redundant `.ToBsonDocument()` right after pushing the code, so I force pushed. Could you please recheck?

**Review by vincentkam — APPROVED** [review]
LGTM! 😸

File: tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs [hunk 2]
```diff
@@ -89,7 +89,14 @@ public void CreateIndex_with_wildcardProjection_should_create_expected_index()
             var indexes = collection.GetIndexes();
             var index = indexes.RawDocuments.Single(i => i["name"].AsString == "custom");
             index["key"]["$**"].AsInt32.Should().Be(1);
-            index["wildcardProjection"].ToBsonDocument().Should().Be(BsonDocument.Parse("{ b : 1, _id : 0 }"));
+            if (CoreTestConfiguration.ServerVersion >= new SemanticVersion(4, 5, 0, ""))
+            {
+                index["wildcardProjection"].Should().Be(BsonDocument.Parse("{ b : true, _id : false }"));
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs:94]: nit: `index["wildcardProjection"]` this part also can be moved to the line before `if`

Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs:94]: nit2: or even we can have code like this:

    BsonValue expectedValue = cond ? (BsonValue)true: (BsonValue)1; 
    index["wildcardProjection"].Should().Be(BsonDocument.Parse("{ b : true, _id : {expectedValue} }"));

**Review by DmitryLukyanov — APPROVED** [review]

**Review by jyemin — APPROVED** [review]
LGTM

File: tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs (hunk 2 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTypedTests.cs:94]: I'd say that separate lines look easier to read, so I would rather leave them as is.

**Pull request closed without merging**